editor/connections_dialog.cpp

```diff
@@ -1203,58 +1203,68 @@ void ConnectionsDock::_slot_menu_about_to_popup() {
 }
 
 void ConnectionsDock::_tree_gui_input(const Ref<InputEvent> &p_event) {
-	// Handle Delete press.
-	if (ED_IS_SHORTCUT("connections_editor/disconnect", p_event)) {
-		TreeItem *item = tree->get_selected();
-		if (item && _get_item_type(*item) == TREE_ITEM_TYPE_CONNECTION) {
-			Connection connection = item->get_metadata(0);
-			_disconnect(connection);
-			update_tree();
+	const Ref<InputEventKey> &key = p_event;
+
+	if (key.is_valid() && key->is_pressed() && !key->is_echo()) {
+		// Handle Delete press.
+		if (ED_IS_SHORTCUT("connections_editor/disconnect", p_event)) {
```

Comment:
```suggestion
		if (ED_IS_SHORTCUT("connections_editor/disconnect", p_event)) { // Handle Delete press.
```

Reply:
Honestly, I would probably get rid of both comment lines you've highlighted. I'm not sure they add anything. It's already clear what we're handling from the code line itself.

Reply:
I'd agree

Reply:
Ah, I already pushed these suggestions. Should I delete them?